optuna/storages/_rdb/storage.py

```diff
@@ -462,11 +462,57 @@ def _create_new_trial(
 
         """
 
-        session = self.scoped_session()
+        # Retry a couple of times. Deadlocks may occur in distributed environments.
+        n_retries = 0
+        while True:
+            session = self.scoped_session()
```

Comment:
Just a clarification, but do we have to re-create session instances when we retry?

Reply:
We can create one single session and reuse it over multiple transactions. I can change it if you prefer that. In my understand we don't have that well established norms for the lifetime of sessions.

Reply:
Thank you for your reply. 

>  we don't have that well established norms for the lifetime of sessions.

I see. This topic seems to be out of the scope of this PR, so please keep the implementation in this PR.